azavea/cac-tripplanner
# Feature/kak/separate tour markers style#1149

## Overview

Give tour itineraries different styling from user-draggable itineraries with waypoints. Also style markers for event map page and highlight place marker on sidebar card hover.

Style tour itineraries on map hover to highlight the segment between the preceding and next tour destinations, and interact with the sidebar to highlight the next waypoint card.


### Demo

Tour map page (no hover):
![image](https://user-images.githubusercontent.com/960264/67309309-a0822b80-f4c9-11e9-8be5-432f372ef945.png)


Tour place card hover:
![image](https://user-images.githubusercontent.com/960264/67309016-28b40100-f4c9-11e9-849e-3a5c0126ceda.png)


First tour place card hover, origin set:
![image](https://user-images.githubusercontent.com/960264/67309195-6ca70600-f4c9-11e9-8afc-6e1322e750f6.png)


Event page, sidebar hover (marker highlighted):
![image](https://user-images.githubusercontent.com/960264/67309460-e7702100-f4c9-11e9-9031-d1a828190d94.png)


## Notes

The [Leaflet awesome markers library](https://github.com/lvoogdt/Leaflet.awesome-markers) we're using for the map markers have a limited set of colors that can be used for the background for the default icon. One of the two purple colors available happens to be close to the app purple, so I went with that for the highlight marker color. The wireframes show the non-highlighted marker color to match the grey of non-highlighted route segments; for that I went with black instead. The contrast between the black and purple colors is pretty low.


## Testing Instructions

 * Go to the map page for a tour
 * Place card hover interactions should work as expected
 * Itinerary hover interactions should match place card hover interactions
 * Go to the map page for an event
 * Place card hover should highlight place map marker

Closes #1149
